.github/workflows/build-docker.yml

```
@@ -0,0 +1,111 @@
+name: Docker Build
+on:
+  push:
+    branches: [main, 'release-v**']
```

I think we can replace the `release-*` branch with tags which only happen for releases and reduce the number of docker images built:
```suggestion
    branches: [main]
    tags:
      - 'v*'
```